src/Composer/Installer.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\ManagerPlugin\Composer;
+
+use Composer\Package\Locker;
+use Composer\Package\RootPackageInterface;
+use Contao\ManagerPlugin\Dependency\DependencyResolverTrait;
+use Contao\ManagerPlugin\Dependency\DependentPluginInterface;
+use Contao\ManagerPlugin\Dependency\UnresolvableDependenciesException;
+
+class Installer
+{
+    use DependencyResolverTrait;
+
+    private static $generatedClassTemplate = <<<'PHP'
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\ManagerPlugin;
+
+/**
+ * This class is generated by contao/manager-bundle, specifically by
+ * @see \Contao\ManagerBundle\Composer\Installer
+ *
+ * This file is overwritten at every run of `composer install` or `composer update`.
+ */
+%s
+{
+    const BUNDLE_PLUGINS = 'Contao\ManagerPlugin\Bundle\BundlePluginInterface';
+    const CONFIG_PLUGINS = 'Contao\ManagerPlugin\Config\ConfigPluginInterface';
+    const EXTENSION_PLUGINS = 'Contao\ManagerPlugin\Config\ExtensionPluginInterface';
+    const ROUTING_PLUGINS = 'Contao\ManagerPlugin\Routing\RoutingPluginInterface';
+
+    /**
+     * @var array
+     */
+    private $plugins;
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
+    {
+        $this->plugins = %s;
```

Comment:
This seems unnecessary. You might as well add the plugins directly in the class property (which should be static BTW) like in `ocramius/package-versions`.

Reply:
You can't have objects in a constant, and a constant cannot be initialized afterwards.

Reply:
I see.